exasol/bucketfs/_path.py

```diff
@@ -578,5 +579,125 @@ def build_path(**kwargs) -> PathLike:
+
+
+def get_database_id_by_name(
+    host: str, account_id: str, pat: str, database_name: str
+) -> str:
+    database_id = get_database_id(
+        host=host, account_id=account_id, pat=pat, database_name=database_name
+    )
+    if not database_id:
+        raise ValueError(f"Could not find database_id for name {database_name}")
+    return database_id
+
+
+def infer_path(
+    bucketfs_host: str | None = None,
+    bucketfs_port: int | None = None,
+    bucketfs_name: str | None = None,
+    bucket: str | None = None,
+    bucketfs_user: str | None = None,
+    bucketfs_password: str | None = None,
+    bucketfs_use_https: bool = True,
+    saas_url: str | None = None,
+    saas_account_id: str | None = None,
+    saas_database_id: str | None = None,
+    saas_database_name: str | None = None,
+    saas_token: str | None = None,
+    path_in_bucket: str = "",
+    use_ssl_cert_validation: bool = True,
+    ssl_trusted_ca: str | None = None,
+) -> PathLike | None:
+    """
+    Infers the correct storage backend (on-premises BucketFS or SaaS) from the provided parameters
+    and returns a PathLike object for accessing the specified resource.
+
+    Raises:
+        ValueError: If the parameters are insufficient or inconsistent and the backend cannot be determined.
+    """
+    backend = infer_backend(
+        bucketfs_host,
+        bucketfs_port,
+        bucketfs_name,
+        bucket,
+        bucketfs_user,
+        bucketfs_password,
+        saas_url,
+        saas_account_id,
+        saas_database_id,
+        saas_database_name,
+        saas_token,
+    )
+    if backend == "onprem":
+        bfs_url = f"{'https' if bucketfs_use_https else 'http'}://{bucketfs_host}:{bucketfs_port}"
+        verify = ssl_trusted_ca or use_ssl_cert_validation
+        return build_path(
+            backend=StorageBackend.onprem,
+            url=bfs_url,
+            username=bucketfs_user,
+            password=bucketfs_password,
+            service_name=bucketfs_name,
+            bucket_name=bucket,
+            verify=verify,
+            path=path_in_bucket,
+        )
+    elif backend == "saas":
+        if not saas_database_id and saas_database_name:
+            saas_database_id = get_database_id_by_name(
+                host=saas_url,
+                account_id=saas_account_id,
+                pat=saas_token,
+                database_name=saas_database_name,
+            )
+        elif not saas_database_id and not saas_database_name:
+            raise ValueError(
+                "Incomplete parameter list. "
+                "Please either provide saas_database_id or saas_database_name."
+            )
+        return build_path(
+            backend=StorageBackend.saas,
+            url=saas_url,
+            account_id=saas_account_id,
+            database_id=saas_database_id,
+            pat=saas_token,
+            path=path_in_bucket,
+        )
+    else:
```

Comment:
Sorry, for a late comment. I think it might be useful to also include the Mounted BucketFS path here (and in the `infer_backend`). This backend is just a file path that can be used by either a UDF or for testing. See `build_path`.
The ValueError would probably be "Unsupported backend". Insufficient parameters would be thrown by the `infer_backend`. If it was happy with the parameters, this must be some kind of a new backend, this function is unaware of or really doesn't want to support. This is a very small thing of course.